event_routing_backends/processors/xapi/event_transformers/enrollment_events.py

```diff
@@ -0,0 +1,90 @@
+"""
+Transformers for enrollment related events.
+"""
+from logging import getLogger
+
+from django.conf import settings
+from django.urls import reverse
+from tincan import Activity, ActivityDefinition, Context, Extensions, LanguageMap, Verb
+
+# from event_routing_backends.helpers import get_anonymous_user_id_by_username
+from event_routing_backends.helpers import get_course_from_id
+from event_routing_backends.processors.xapi import constants
+from event_routing_backends.processors.xapi.registry import XApiTransformersRegistry
+from event_routing_backends.processors.xapi.transformer import XApiTransformer
+
+logger = getLogger(__name__)
+
+
+class BaseEnrollmentTransformer(XApiTransformer):
+    """
+    Base transformer for enrollment events.
+    """
+    additional_fields = ('context', )
+
+    def get_object(self):
+        """
+        Get object for xAPI transformed event.
+
+        Returns:
+            `Activity`
+        """
+        course_id = self.find_nested('course_id')
+        # TODO: make a method to get course URL. and use that everywhere
+        object_id = '{root_url}{course_root_url}'.format(
+                    root_url=settings.LMS_ROOT_URL,
+                    course_root_url=reverse('course_root', kwargs={
+                        'course_id': course_id
+                    })
+        )
+
+        course = get_course_from_id(course_id)
```

Comment:
Why do we need the full course object here?  Can you tell me more about where/how it gets used?

Reply:
here we are extracting the course's display name in which the user is being enrolled

Reply:
This could be done using the course overview object instead which should be a lot faster and cheaper. https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/api.py#L12